docs/simplify-predicate.md

```diff
@@ -0,0 +1,88 @@
+# Simplifying predicates
+
+The goals of simplification are:
+
+1. Transform the predicate into [disjunctive normal form] (normalization).
+2. Apply identities to reduce the complexity of the predicate (simplification proper).
+
+Simplification happens recursively, upward from the bottom. Therefore, we
+describe the simplification steps based on the shape of the top-most
+unsimplified clause of the predicate.
+
+## \and
+
+### Normalization
+
+Apply this distributive law over both arguments of `\and` until `P[n]` are all
+in disjunctive normal form:
+
+``` kore
+\and(\or(P[1], P[2]), P[3]) = \or(\and(P[1], P[3]), \and(P[2], P[3]))
+```
+
+### Simplification
+
+#### Identity
+
+``` kore
+\and(\top, P[1]) = P[1]
+```
+
+#### Annihilation
+
+``` kore
+\and(\bottom, _) = \bottom
+```
+
+#### Idempotence
+
+``` kore
+\and(P[1], P[1]) = P[1]
+```
+
+#### Affirmation
+
+Idempotence is actually a special case of _affirmation_,
+
+``` kore
+\and(P[1], P[2]) = \and(P[1], P[2](P[1] = \top))
+```
+
+## \or
+
+### Normalization
+
+Assuming the children of `\or` are in disjunctive normal form, then the clause
+is already normalized.
```

Comment:
Maybe add something like "because `or` is associative", to anticipate `normalizeOr` better.